pilot/pkg/model/sidecar.go

```diff
@@ -372,6 +372,20 @@ func (ilw *IstioEgressListenerWrapper) VirtualServices() []Config {
 	return ilw.virtualServices
 }
 
+// ContainsEgressNamespace determines if the Sidecar includes the given namespace.
+func (ilw *IstioEgressListenerWrapper) ContainsEgressNamespace(namespace string) bool {
+	if ilw == nil {
+		return true
+	}
+
+	for hostNs := range ilw.listenerHosts {
+		if hostNs == wildcardNamespace || namespace == hostNs {
```

Comment:
So using listenerHosts is not the right logic here as its going to lead to confusions like the wildcard namespace above. What you can do is add an array of namespace strings (pre-process along with list of Services[]). Then this function will become a simple map check ilw.namespaces[inNamespace] [ O(1) ] and not have to iterate through all the hosts in listener hosts.

Reply:
To be even more efficient, you shouldn't be storing this in the egressListenerWrapper. Just compute the set of dependent namespaces and store it in the parent SidecarScope object. this will avoid the loop in ads.go

Reply:
sgtm